src/Compilers/Core/Portable/Compilation/Expression.cs

```diff
@@ -81,401 +75,4 @@ public static BinaryOperationKind DeriveAdditionKind(ITypeSymbol type)
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitBinaryOperatorExpression(this, argument);
-        }
-    }
-
-    internal sealed class ArrayCreation : IArrayCreationExpression
-    {
-        private readonly IArrayTypeSymbol _arrayType;
-
-        public ArrayCreation(IArrayTypeSymbol arrayType, ImmutableArray<IOperation> elementValues, SyntaxNode syntax)
-        {
-            _arrayType = arrayType;
-            this.DimensionSizes = ImmutableArray.Create<IOperation>(new IntegerLiteral(elementValues.Count(), null, syntax));
-            this.Initializer = new ArrayInitializer(elementValues, syntax, arrayType);
-            this.Syntax = syntax;
-        }
-
-        public ITypeSymbol Type => _arrayType;
-
-        public ImmutableArray<IOperation> DimensionSizes { get; }
-
-        public ITypeSymbol ElementType => _arrayType.ElementType;
-
-        public IArrayInitializer Initializer { get; }
-
-        public SyntaxNode Syntax { get; }
-
-        public OperationKind Kind => OperationKind.ArrayCreationExpression;
-
-        public bool IsInvalid => IsInvalidInitializer(Initializer);
-
-        private static bool IsInvalidInitializer(IArrayInitializer initializer) => initializer.IsInvalid;
-
-        public void Accept(OperationVisitor visitor)
-        {
-            visitor.VisitArrayCreationExpression(this);
-        }
-
-        public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-        {
-            return visitor.VisitArrayCreationExpression(this, argument);
-        }
-
-        public Optional<object> ConstantValue => default(Optional<object>);
-
-        private sealed class ArrayInitializer : IArrayInitializer
-        {
-            public ArrayInitializer(ImmutableArray<IOperation> elementValues, SyntaxNode syntax, ITypeSymbol arrayType)
-            {
-                ElementValues = elementValues;
-                Syntax = syntax;
-                Type = arrayType;
-            }
-
-            public ImmutableArray<IOperation> ElementValues { get; }
-
-            public bool IsInvalid => ElementValues.Any(v => v.IsInvalid);
-
-            public OperationKind Kind => OperationKind.ArrayInitializer;
-
-            public ITypeSymbol Type { get; }
-
-            public SyntaxNode Syntax { get; }
-
-            public Optional<object> ConstantValue => default(Optional<object>);
-
-            public void Accept(OperationVisitor visitor)
-            {
-                visitor.VisitArrayInitializer(this);
-            }
-
-            public TResult Accept<TArgument, TResult>(OperationVisitor<TArgument, TResult> visitor, TArgument argument)
-            {
-                return visitor.VisitArrayInitializer(this, argument);
-            }
-        }
-    }
-
-    internal sealed class InvalidExpression : IInvalidExpression
```

Comment:
the existing file contained ones that are related to IOperations and one that didn't related to IOperations. so I split the file and put ones that related to IOperations in Operations folder with "Expressions.cs"